fea-rs/src/compile/validate.rs

```diff
@@ -207,6 +213,25 @@ impl<'a> ValidationCtx<'a> {
         }
     }
 
+    fn validate_condition_set(&mut self, node: &typed::ConditionSet) {
+        let label = node.label().to_owned();
+        if let Some(_prev) = self.condition_set_defs.insert(label.text.clone(), label) {
+            self.warning(node.label().range(), "duplicate condition set definition");
+        }
+    }
+
+    fn validate_feature_variation(&mut self, node: &typed::FeatureVariation) {
+        let feature_tag = node.tag();
+        if let Some(cond_set) = node.condition_set() {
+            if !self.condition_set_defs.contains_key(cond_set.as_str()) {
+                self.error(cond_set.range(), "undefined conditionset");
+            }
+        } else {
+            assert!(node.null().is_some(), "go fix your parser");
```

Comment:
Is this meant to mean fea-rs has a bug if you see this?

Reply:
yes, it would mean that there's a major regression in the parser, and we have tests for this.